### PR TITLE
Stack, Queue, Linkedlist implementations

### PR DESCRIPTION
**Pull request by yvnagapreethi**

### CODE REVIEW

File: doublelinkedlist.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+"""Class code goes from here"""
```
Comment (mithileshkunal) [inline doublelinkedlist.py]: Remove this.

File: exmps.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+color_list = ["Red", "Green", "White", "Black"]
```
Comment (mithileshkunal) [inline exmps.py:2]: What is the purpose of this program?
Suggestion: Write comments for complex programs to explain the problem statement

File: methods.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    @staticmethod
+    def static_method():
+        print("static method")
+
+    def inst_methond(self):
+        print("calling instance method")
+
+
+MethodsOfPython.static_method()
```
Comment (mithileshkunal) [inline methods.py]: Add __name__ == "__main__" to run this code.

File: printnames.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class PrintNames:
+
+    def __init__(self):
+        first = input("Enter your first name")
+        last = input("Enter your last name")
+        print("Final is {0} {1}".format(first[-1::-1], last[-1::-1]))
```
Comment (mithileshkunal) [inline printnames.py]: `Full name` instead of `Final`?

File: printnames.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class PrintNames:
+
+    def __init__(self):
```
Comment (mithileshkunal) [inline printnames.py]: Modify the constructor to pass the values as arguments.

**Review by mithileshkunal — CHANGES_REQUESTED** [review]
Added my comments.

File: doublelinkedlist.py (hunk 1 above)
Comment (yvnagapreethi) [reply doublelinkedlist.py]: Removed and updated

File: methods.py (hunk 3 above)
Comment (yvnagapreethi) [reply methods.py]: Updated

File: printnames.py (hunk 5 above)
Comment (yvnagapreethi) [reply printnames.py]: Made changes

File: printnames.py (hunk 4 above)
Comment (yvnagapreethi) [reply printnames.py]: Done

File: areaofcircle.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class AreaOfCirlce:
```
Comment (mithileshkunal) [inline areaofcircle.py]: As discussed, modify this to create Circle class.

File: areaofcircle.py [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+    def __init__(self):
+        self.radius = float(input("Please enter the radius value to find the area of a cirlce"))
+        print("The area of a circle is:{0} ".format(self.area1(self.radius)))
+
+    def area1(self,radius):
+        return 3.14*self.radius**2
+        # print("The radius is {0}".format(radius))
+
+
+ar = AreaOfCirlce()
```
Comment (mithileshkunal) [inline areaofcircle.py]: If you are not using the object `ar` later, create the object without assigning it to variable.

File: printnames.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class PrintNames:
+
+    def __init__(self, first_name, last_name):
+        self.first_name = first_name
+        self.last_name = last_name
+        print("Full Name is {0} {1}".format(self.first_name[-1::-1], self.last_name[-1::-1]))
```
Comment (mithileshkunal) [inline printnames.py]: Don't print here. Add code to return this when the object is printed.

**Review by mithileshkunal — COMMENTED** [review]
Added more comments

File: areaofcircle.py (hunk 6 above)
Comment (yvnagapreethi) [reply areaofcircle.py]: Changed the class name

File: printnames.py (hunk 8 above)
Comment (yvnagapreethi) [reply printnames.py]: done

**Review by mithileshkunal — APPROVED** [review]

File: areaofcircle.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+    def __init__(self):
+        self.radius = float(input("Please enter the radius value to find the area of a cirlce"))
+        print("The area of a circle is:{0} ".format(self.area1(self.radius)))
+
+    def area1(self, radius):
+        return 3.14 * self.radius**2
+        # print("The radius is {0}".format(radius))
+
+
```
Comment (mithileshkunal) [inline areaofcircle.py]: if __name__ == "__main__":

File: exmps.py [hunk 10]
```diff
@@ -0,0 +1,17 @@
+"""Practicised few questions from google"""
```
Comment (mithileshkunal) [inline exmps.py]: `"""` should be used only for docstrings. For all other comments use `#`

File: printnames.py [hunk 11]
```diff
@@ -0,0 +1,15 @@
+class PrintNames:
+
+    def __init__(self, first_name, last_name):
+        self.first_name = first_name
+        self.last_name = last_name
+        print(self.print_complete_name())
```
Comment (mithileshkunal) [inline printnames.py:6]: Remove this here.

File: printnames.py [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+    def print_complete_name(self):
+        return "Full Name is {0} {1}".format(self.first_name[-1::-1], self.last_name[-1::-1])
+
+
+if __name__ == '__main__':
+    first_name = input("Enter the first name")
+    last_name = input("Enter the last name")
+    pn = PrintNames(first_name, last_name)
```
Comment (mithileshkunal) [inline printnames.py:15]: print this object pn

File: stackusinglinkedlist.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+    def push_stack_element(self,data):
+        if self.root is None:
+            new = StackNode(data)
+            self.root = new
+        else:
+            new_node = StackNode(data)
+            new_node.next = self.root
+            self.root = new_node
+
+    def pop_stack_element(self):
```
Comment (mithileshkunal) [inline stackusinglinkedlist.py:20]: `pop` should return the value to the caller.

File: stackusinglist.py [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+            print("There are no elments in the stack to pop")
+        else:
+            print("The poped element is: {}".format(self.stack[-1]))
+            self.stack.pop()
+
+
+if __name__ == "__main__":
+    stk = StackUsingList()
+    while True:
+        print("Select any number from below:\n1.Push an element into the stack\n2.Pop out an element from the stack\n3.Display all the elements present in the stack\n")
```
Comment (mithileshkunal) [inline stackusinglist.py:24]: Add all these messages in a list and print from that list

**Review by mithileshkunal — COMMENTED** [review]
Added comments for stack and linkedlist

**Pull request merged**